tools/cldr-code/src/main/java/org/unicode/cldr/util/BoundaryTransform.java

```diff
@@ -0,0 +1,219 @@
+package org.unicode.cldr.util;
```

Comment:
Should this be in an existing or new icu class (since presumably it will go there eventually)?

Reply:
Once we get this further along, it (or an optimized version of it) could go into ICU. We should work first on the rules, and make sure that the syntax is powerful enough for what we need (and reasonably clear).

Reply:
Note that the current implementation uses regex for convenience. We would need to make sure that we use syntax that is easily supported across platforms. It doesn't have to be the same, but would have to be easily transformed into "native" regex.